include/deal.II/matrix_free/tensor_product_kernels.h

```diff
@@ -1263,49 +1252,13 @@ namespace internal
               bool contract_over_rows,
               bool add,
               bool one_line     = false,
-              EvaluatorQuantity = EvaluatorQuantity::value>
+              EvaluatorQuantity = EvaluatorQuantity::value,
+              int extra_stride  = 1>
     static void
     apply(const Number2 *DEAL_II_RESTRICT shape_data,
           const Number                   *in,
           Number                         *out);
 
-    /**
-     * This function applies the tensor product operation to produce face values
-     * from cell values. As opposed to the apply method, this method assumes
-     * that the directions orthogonal to the face have n_rows degrees of
-     * freedom per direction and not n_columns for those directions lower than
-     * the one currently applied. In other words, apply_face() must be called
-     * before calling any interpolation within the face.
-     *
-     * @tparam face_direction Direction of the normal vector (0=x, 1=y, etc)
-     * @tparam contract_onto_face If true, the input vector is of size n_rows^dim
-     *                            and interpolation into n_rows^(dim-1) points
-     *                            is performed. This is a typical scenario in
-     *                            FEFaceEvaluation::evaluate() calls. If false,
-     *                            data from n_rows^(dim-1) points is expanded
-     *                            into the n_rows^dim points of the higher-
-     *                            dimensional data array. Derivatives in the
-     *                            case contract_onto_face==false are summed
-     *                            together
-     * @tparam add If true, the result is added to the output vector, else
-     *             the computed values overwrite the content in the output
-     * @tparam max_derivative Sets the number of derivatives that should be
-     *             computed. 0 means only values, 1 means values and first
-     *             derivatives, 2 second derivates. Note that all the
-     *             derivatives access the data in @p shape_values passed to
-     *             the constructor of the class
-     *
-     * @param in address of the input data vector
-     * @param out address of the output data vector
-     */
-    template <int  face_direction,
-              bool contract_onto_face,
-              bool add,
-              int  max_derivative>
-    void
-    apply_face(const Number *DEAL_II_RESTRICT in,
-               Number *DEAL_II_RESTRICT       out) const;
```

Comment:
Note that this function is completely gone from `tensor_product_kernels`. It is replaced by the free function `interpolate_to_face` that is sufficiently general to support both the regular evaluators as well as Raviart-Thomas (through a mix of template parameters and run-time parameters).